@ghalliday please review latest changes. 
Also, failures do not seem to be related to the commits:

 #27 2.597 CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
#27 2.597 CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
#27 2.597 CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage

--- a/common/remote/rmtspawn.cpp
+++ b/common/remote/rmtspawn.cpp
@@ -417,7 +417,7 @@ void CRemoteSlave::run(int argc, const char * * argv)
     logFile.append(slaveName);
     addFileTimestamp(logFile, true);
     logFile.append(".log");
-    attachStandardFileLogMsgMonitor(logFile.str(), 0, MSGFIELD_STANDARD, MSGAUD_all, MSGCLS_all, TopDetail, false, true, true);
+    attachStandardFileLogMsgMonitor(logFile.str(), 0, MSGFIELD_STANDARD, MSGAUD_all, MSGCLS_all, TopDetail, LOGFORMAT_table, true, true);
     queryLogMsgManager()->removeMonitor(queryStderrLogMsgHandler());        // no point logging output to screen if run remote!
     LOG(MCdebugProgress, unknownJob, "Starting %s %s %s %s %s %s %s",slaveName.get(),(argc>1)?argv[1]:"",(argc>2)?argv[2]:"",(argc>3)?argv[3]:"",(argc>4)?argv[4]:"",(argc>5)?argv[5]:"",(argc>6)?argv[6]:"");
 #else

--- a/dali/daliadmin/daliadmin.cpp
+++ b/dali/daliadmin/daliadmin.cpp
@@ -180,7 +180,7 @@ int main(int argc, const char* argv[])
         Owned<ILogMsgHandler> fileMsgHandler;
         if (props->getProp("logfile",logname)) {
             if (logname.length()) {
-                fileMsgHandler.setown(getFileLogMsgHandler(logname.str(), NULL, rawlog?MSGFIELD_prefix:MSGFIELD_STANDARD, false, false, true));
+                fileMsgHandler.setown(getFileLogMsgHandler(logname.str(), NULL, rawlog?MSGFIELD_prefix:MSGFIELD_STANDARD, LOGFORMAT_table, false, true));
                 queryLogMsgManager()->addMonitorOwn(fileMsgHandler.getClear(), getCategoryLogMsgFilter(MSGAUD_all, MSGCLS_all, TopDetail));
             }
         }

--- a/dali/sasha/sasha.cpp
+++ b/dali/sasha/sasha.cpp
@@ -407,7 +407,7 @@ int main(int argc, char* argv[])
     Thread::setDefaultStackSize(0x10000);
     try {
         startMPServer(0);
-        attachStandardFileLogMsgMonitor("sasha.log", NULL, MSGFIELD_STANDARD, MSGAUD_all, MSGCLS_all, TopDetail, false, true);
+        attachStandardFileLogMsgMonitor("sasha.log", NULL, MSGFIELD_STANDARD, MSGAUD_all, MSGCLS_all, TopDetail, LOGFORMAT_table, true);
         queryStderrLogMsgHandler()->setMessageFields(MSGFIELD_prefix);
 
         SocketEndpoint ep;

--- a/ecl/eclcc/eclcc.cpp
+++ b/ecl/eclcc/eclcc.cpp
@@ -3213,7 +3213,7 @@ void EclCC::processBatchedFile(IFile & file, bool multiThreaded)
         {
             if (!multiThreaded)
             {
-                handler.setown(getHandleLogMsgHandler(logFile, 0, false));
+                handler.setown(getHandleLogMsgHandler(logFile, 0, LOGFORMAT_table));
                 Owned<ILogMsgFilter> filter = getCategoryLogMsgFilter(MSGAUD_all, MSGCLS_all, DefaultDetail);
                 queryLogMsgManager()->addMonitor(handler, filter);
 

--- a/fs/dafscontrol/dafscontrol.cpp
+++ b/fs/dafscontrol/dafscontrol.cpp
@@ -282,7 +282,7 @@ int main(int argc, char* argv[])
     }
 
     EnableSEHtoExceptionMapping();
-    attachStandardFileLogMsgMonitor("dafscontrol.log", NULL, MSGFIELD_STANDARD, MSGAUD_all, MSGCLS_all, TopDetail, false);
+    attachStandardFileLogMsgMonitor("dafscontrol.log", NULL, MSGFIELD_STANDARD, MSGAUD_all, MSGCLS_all, TopDetail);
     queryStderrLogMsgHandler()->setMessageFields(MSGFIELD_prefix);
     int ret = 0;
     try {

--- a/helm/examples/logging/README.md
+++ b/helm/examples/logging/README.md
@@ -65,11 +65,11 @@ This configuration coincides with the elastic4hpcclogs managed solution found in
 
 As mentioned earlier, the HPCC Systems logs provide a wealth of information which can be used for benchmarking, auditing, debugging, monitoring, etc. The type of information provided in the logs and its format is trivially controlled via standard Helm configuration.
 
-By default, the component logs are not filtered, and contain the following columns:
+By default, the component logs are not filtered, are reported in a space delimited values format, and contain the following columns:
     
     MessageID TargetAudience LogEntryClass JobID DateStamp TimeStamp ProcessId ThreadID QuotedLogMessage
 
-The logs can be filtered by TargetAudience, Category or Detail Level, and the output columns can be configured. Logging configuration settings can be applied at the global, or component level.
+The logs can be filtered by TargetAudience, Category or Detail Level, and the output columns can be configured. The logs are reported in 'table' (space delimited values) format by default, and can be re-configured to XML or JSON format. Logging configuration settings can be applied at the global, or component level.
 
 ### Target Audience Filtering
 
@@ -116,3 +116,45 @@ NB: normally it is expected that the logging stack will keep up and the default 
 The defaults can be configured by setting `<section>`.logging.queueLen and/or `<section>`.logging.queueDrop.
 Setting `<section>`.logging.queueLen to 0, will disabled asynchronous logging, i.e. each log will block until completed.
 Setting `<section>`.logging.queueDrop to a non-zero (N) value will cause N logging entries from the queue to be discarded if the queueLen is reached.
+
+### Log data format configuration
+
+By default, HPCC Systems component logs are reported in the format known as 'table' which consists of space delimited value columns.
+
+    For example:
+    0000CF0F PRG INF 2020-05-12 17:10:34.910     1 10690 "HTTP First Line: GET / HTTP/1.1"
+    0000CF10 PRG INF 2020-05-12 17:10:34.911     1 10690 "GET /, from 10.240.0.4"
+    0000CF11 PRG INF 2020-05-12 17:10:34.911     1 10690 “TxSummary[activeReqs=22; rcv=5ms;total=6ms;]"
+
+The log format can be re-configured to xml or json. To do so, specify the desired format in the logging.format section:
+
+Sample configuration for json format:
+
+      logging:
+        format: json
+
+Sample json formated log output:
+
+    { "MSG": "HTTP connection from 1.1.1.1:52392 on persistent socket", "MID": "593", "AUD": "PRG", "CLS": "INF", "DATE": "2023-02-23", "TIME": "17:42:42.808", "PID": "8", "TID": "117", "JOBID": "UNK" }
+    { "MSG": "HTTP First Line: POST /WsWorkunits/WUQuery.json HTTP/1.1", "MID": "594", "AUD": "PRG", "CLS": "INF", "DATE": "2023-02-23", "TIME": "17:42:42.809", "PID": "8", "TID": "225", "JOBID": "UNK" }
+    { "MSG": "TxSummary[activeReqs=1;auth=NA;contLen=31;rcv=1;handleHttp=9;req=POST wsworkunits.WUQUERY v1.95;total=9;]", "MID": "592", "AUD": "PRG", "CLS": "INF", "DATE": "2023-02-23", "TIME": "17:42:39.277", "PID": "8", "TID": "218", "JOBID": "UNK" }
+
+Sample configuration for xml format:
+
+      logging:
+        format: xml
+
+Sample xml formatted log output:
+
+    <msg MessageID="150" Audience="Programmer" Class="Warning"
+     date="2023-02-23" time="17:47:33.516" PID="8" TID="114"
+     JobID="UNK"
+     text="TxSummary entry with name 'rcv' and suffix 'ms'; suffix is ignored" />
+    <msg MessageID="151" Audience="Programmer" Class="Information"
+     date="2023-02-23" time="17:47:33.516" PID="8" TID="114"
+     JobID="UNK"
+     text="POST /WsWorkunits/WUDetails.json, from 1.1.1.41" />
+    <msg MessageID="152" Audience="User" Class="Progress"
+     date="2023-02-23" time="17:47:33.517" PID="8" TID="114"
+     JobID="UNK"
+     text="WUDetails: W20230223-174728" />

--- a/helm/hpcc/values.schema.json
+++ b/helm/hpcc/values.schema.json
@@ -1017,6 +1017,11 @@
           "type": "integer",
           "description": "Number of log entries to drop from the log queue when it is full. Disabled if 0, or if queueLength is 0",
           "default": 0
+        },
+        "dataFormat": {
+          "type": "string",
+          "description": "The data format used to report logging: xml|json|table(space delimited values)",
+          "enum": [ "xml", "json", "table"]
         }
       },
       "additionalProperties": { "type": ["integer", "string", "boolean"] }

--- a/system/jlib/jlog.cpp
+++ b/system/jlib/jlog.cpp
@@ -45,7 +45,7 @@ static ILogMsgManager * theManager = nullptr;
 static PassAllLogMsgFilter * thePassAllFilter = nullptr;
 static PassLocalLogMsgFilter * thePassLocalFilter = nullptr;
 static PassNoneLogMsgFilter * thePassNoneFilter = nullptr;
-static HandleLogMsgHandlerTable * theStderrHandler = nullptr;
+static HandleLogMsgHandler * theStderrHandler = nullptr;
 static CSysLogEventLogger * theSysLogEventLogger = nullptr;
 
 
@@ -485,6 +485,79 @@ StringBuffer & LogMsg::toStringXML(StringBuffer & out, unsigned fields) const
     return out;
 }
 
+StringBuffer & LogMsg::toStringJSON(StringBuffer & out, unsigned fields) const
+{
+    out.ensureCapacity(LOG_MSG_FORMAT_BUFFER_LENGTH);
+    StringBuffer encodedMsg;
+    out.append("{ \"MSG\": \"").append(encodeJSON(encodedMsg, text.str())).append("\"");
+
+    if(fields & MSGFIELD_msgID)
+        out.append(", \"MID\": \"").append(sysInfo.queryMsgID()).append("\"");
+    if(fields & MSGFIELD_audience)
+        out.append(", \"AUD\": \"").append(LogMsgAudienceToFixString(category.queryAudience())).append("\"");
+    if(fields & MSGFIELD_class)
+        out.append(", \"CLS\": \"").append(LogMsgClassToFixString(category.queryClass())).append("\"");
+    if(fields & MSGFIELD_detail)
+        out.append(", \"DET\": \"").append(category.queryDetail()).append("\"");
+    if(fields & MSGFIELD_timeDate)
+    {
+        time_t timeNum = sysInfo.queryTime();
+        char timeString[23];
+        struct tm timeStruct;
+        localtime_r(&timeNum, &timeStruct);
+        if(fields & MSGFIELD_date)
+        {
+            strftime(timeString, 23, ", \"DATE\": \"%Y-%m-%d\"", &timeStruct);
+            out.append(timeString);
+        }
+        if(fields & MSGFIELD_microTime)
+        {
+            out.appendf(", \"TIME\": \"%02d:%02d:%02d.%06d\"", timeStruct.tm_hour, timeStruct.tm_min, timeStruct.tm_sec, sysInfo.queryUSecs());
+        }
+        else if(fields & MSGFIELD_milliTime)
+        {
+            out.appendf(", \"TIME\": \"%02d:%02d:%02d.%03d\"", timeStruct.tm_hour, timeStruct.tm_min, timeStruct.tm_sec, sysInfo.queryUSecs()/1000);
+        }
+        else if(fields & MSGFIELD_time)
+        {
+            out.appendf(", \"TIME\": \"%02d:%02d:%02d\"", timeStruct.tm_hour, timeStruct.tm_min, timeStruct.tm_sec);
+        }
+    }
+    if(fields & MSGFIELD_process)
+        out.append(", \"PID\": \"").append(sysInfo.queryProcessID()).append("\"");
+    if(fields & MSGFIELD_thread)
+        out.append(", \"TID\": \"").append(sysInfo.queryThreadID()).append("\"");
+    if(fields & MSGFIELD_session)
+    {
+        if(sysInfo.querySessionID() == UnknownSession)
+            out.append(", \"SES\": \"UNK\" ");
+        else
+            out.append(", \"SES\": \"").append(sysInfo.querySessionID()).append("\"");
+    }
+    if(fields & MSGFIELD_node)
+    {
+        out.append(", \"NODE\": \"");
+        sysInfo.queryNode()->getUrlStr(out);
+        out.append("\"");
+    }
+    if(fields & MSGFIELD_job)
+    {
+        out.appendf(", \"JOBID\": \"%s\"", jobInfo.queryJobIDStr());
+    }
+    if(fields & MSGFIELD_user)
+    {
+        if(jobInfo.queryUserID() == UnknownUser)
+            out.append(", \"USER\": \"UNK\" ");
+        else
+            out.append(", \"USER\": \"").append(jobInfo.queryUserID()).append("\"");
+    }
+    if((fields & MSGFIELD_code) && (msgCode != NoLogMsgCode))
+        out.append(", \"CODE\": \"").append(msgCode).append("\"");
+
+    out.append(" }\n");
+    return out;
+}
+
 StringBuffer & LogMsg::toStringTable(StringBuffer & out, unsigned fields) const
 {
     if(fields & MSGFIELD_msgID)
@@ -786,6 +859,25 @@ void HandleLogMsgHandlerTable::addToPTree(IPropertyTree * tree) const
     tree->addPropTree("handler", handlerTree);
 }
 
+void HandleLogMsgHandlerJSON::handleMessage(const LogMsg & msg)
+{
+    CriticalBlock block(crit);
+    msg.toStringJSON(curMsgText.clear(), messageFields);
+    fputs(curMsgText.str(), handle);
+}
+
+void HandleLogMsgHandlerJSON::addToPTree(IPropertyTree * tree) const
+{
+    IPropertyTree * handlerTree = createPTree(ipt_caseInsensitive);
+    if(handle==stderr)
+        handlerTree->setProp("@type", "stderr");
+    else
+        handlerTree->setProp("@type", "mischandle");
+    handlerTree->setPropInt("@fields", messageFields);
+    handlerTree->setProp("@writeJSON", "true");
+    tree->addPropTree("handler", handlerTree);
+}
+
 void HandleLogMsgHandlerXML::handleMessage(const LogMsg & msg)
 {
     CriticalBlock block(crit);
@@ -886,6 +978,31 @@ void FileLogMsgHandlerXML::addToPTree(IPropertyTree * tree) const
     handlerTree->setPropInt("@fields", messageFields);
     if(append) handlerTree->setProp("@append", "true");
     if(flushes) handlerTree->setProp("@flushes", "true");
+    tree->addPropTree("handler", handlerTree);
+}
+
+void FileLogMsgHandlerJSON::handleMessage(const LogMsg & msg)
+{
+    CriticalBlock block(crit);
+    msg.toStringJSON(curMsgText.clear(), messageFields);
+    fputs(curMsgText.str(), handle);
+    if(flushes)
+        fflush(handle);
+}
+
+void FileLogMsgHandlerJSON::addToPTree(IPropertyTree * tree) const
+{
+    IPropertyTree * handlerTree = createPTree(ipt_caseInsensitive);
+    handlerTree->setProp("@type", "file");
+    handlerTree->setProp("@filename", filename.get());
+    if(headerText)
+        handlerTree->setProp("@headertext", headerText.get());
+    handlerTree->setPropInt("@fields", messageFields);
+    handlerTree->setProp("@writeJSON", "true");
+    if(append)
+        handlerTree->setProp("@append", "true");
+    if(flushes)
+        handlerTree->setProp("@flushes", "true");
     tree->addPropTree("handler", handlerTree);
 }
 
@@ -1986,18 +2103,32 @@ ILogMsgFilter * getSwitchLogMsgFilterOwn(ILogMsgFilter * switchFilter, ILogMsgFi
     return ret;
 }
 
-ILogMsgHandler * getHandleLogMsgHandler(FILE * handle, unsigned fields, bool writeXML)
+ILogMsgHandler * getHandleLogMsgHandler(FILE * handle, unsigned fields, LogHandlerFormat logFormat)
 {
-    if(writeXML)
+    switch (logFormat)
+    {
+    case LOGFORMAT_xml:
         return new HandleLogMsgHandlerXML(handle, fields);
-    return new HandleLogMsgHandlerTable(handle, fields);
+    case LOGFORMAT_json:
+        return new HandleLogMsgHandlerJSON(handle, fields);
+    case LOGFORMAT_table:
+    default:
+        return new HandleLogMsgHandlerTable(handle, fields);
+    }
 }
 
-ILogMsgHandler * getFileLogMsgHandler(const char * filename, const char * headertext, unsigned fields, bool writeXML, bool append, bool flushes)
+ILogMsgHandler * getFileLogMsgHandler(const char * filename, const char * headertext, unsigned fields, LogHandlerFormat logFormat, bool append, bool flushes)
 {
-    if(writeXML)
+    switch (logFormat)
+    {
+    case LOGFORMAT_xml:
         return new FileLogMsgHandlerXML(filename, headertext, fields, append, flushes);
-    return new FileLogMsgHandlerTable(filename, headertext, fields, append, flushes);
+    case LOGFORMAT_json:
+        return new FileLogMsgHandlerJSON(filename, headertext, fields, append, flushes);
+    case LOGFORMAT_table:
+    default:
+        return new FileLogMsgHandlerTable(filename, headertext, fields, append, flushes);
+    }
 }
 
 ILogMsgHandler * getRollingFileLogMsgHandler(const char * filebase, const char * fileextn, unsigned fields, bool append, bool flushes, const char *initialName, const char *alias, bool daily, long maxLogSize)
@@ -2034,7 +2165,7 @@ ILogMsgHandler * getLogMsgHandlerFromPTree(IPropertyTree * tree)
             fields = logMsgFieldsFromAbbrevs(fstr);
     }
     if(strcmp(type.str(), "stderr")==0)
-        return getHandleLogMsgHandler(stderr, fields, tree->hasProp("@writeXML"));
+        return getHandleLogMsgHandler(stderr, fields, tree->hasProp("@writeXML") ? LOGFORMAT_xml : LOGFORMAT_table);
     else if(strcmp(type.str(), "file")==0)
     {
         StringBuffer filename;
@@ -2043,10 +2174,14 @@ ILogMsgHandler * getLogMsgHandlerFromPTree(IPropertyTree * tree)
         {
             StringBuffer headertext;
             tree->getProp("@headertext", headertext);
-            return getFileLogMsgHandler(filename.str(), headertext.str(), fields, !(tree->hasProp("@writeTable")), tree->hasProp("@append"), tree->hasProp("@flushes"));
+            //JSON format now available, but currently not an option for file bound logs
+            return getFileLogMsgHandler(filename.str(), headertext.str(), fields, !(tree->hasProp("@writeTable")) ? LOGFORMAT_xml : LOGFORMAT_table, tree->hasProp("@append"), tree->hasProp("@flushes"));
         }
         else
-            return getFileLogMsgHandler(filename.str(), 0, fields, !(tree->hasProp("@writeTable")), tree->hasProp("@append"), tree->hasProp("@flushes"));
+        {
+            //JSON format now available, but currently not an option for file bound logs
+            return getFileLogMsgHandler(filename.str(), 0, fields, !(tree->hasProp("@writeTable")) ? LOGFORMAT_xml : LOGFORMAT_table, tree->hasProp("@append"), tree->hasProp("@flushes"));
+        }
     }
     else if(strcmp(type.str(), "binary")==0)
     {
@@ -2058,13 +2193,13 @@ ILogMsgHandler * getLogMsgHandlerFromPTree(IPropertyTree * tree)
     return LINK(theStderrHandler);
 }
 
-ILogMsgHandler * attachStandardFileLogMsgMonitor(const char * filename, const char * headertext, unsigned fields, unsigned audiences, unsigned classes, LogMsgDetail detail, bool writeXML, bool append, bool flushes, bool local)
+ILogMsgHandler * attachStandardFileLogMsgMonitor(const char * filename, const char * headertext, unsigned fields, unsigned audiences, unsigned classes, LogMsgDetail detail, LogHandlerFormat logFormat, bool append, bool flushes, bool local)
 {
 #ifdef FILE_LOG_ENABLES_QUEUEUING
     queryLogMsgManager()->enterQueueingMode();
 #endif
     ILogMsgFilter * filter = getCategoryLogMsgFilter(audiences, classes, detail, local);
-    ILogMsgHandler * handler = getFileLogMsgHandler(filename, headertext, fields, writeXML, append, flushes);
+    ILogMsgHandler * handler = getFileLogMsgHandler(filename, headertext, fields, logFormat, append, flushes);
     queryLogMsgManager()->addMonitorOwn(handler, filter);
     return handler;
 }
@@ -2080,10 +2215,10 @@ ILogMsgHandler * attachStandardBinLogMsgMonitor(const char * filename, unsigned 
     return handler;
 }
 
-ILogMsgHandler * attachStandardHandleLogMsgMonitor(FILE * handle, unsigned fields, unsigned audiences, unsigned classes, LogMsgDetail detail, bool writeXML, bool local)
+ILogMsgHandler * attachStandardHandleLogMsgMonitor(FILE * handle, unsigned fields, unsigned audiences, unsigned classes, LogMsgDetail detail, LogHandlerFormat logFormat, bool local)
 {
     ILogMsgFilter * filter = getCategoryLogMsgFilter(audiences, classes, detail, local);
-    ILogMsgHandler * handler = getHandleLogMsgHandler(handle, fields, writeXML);
+    ILogMsgHandler * handler = getHandleLogMsgHandler(handle, fields, logFormat);
     queryLogMsgManager()->addMonitorOwn(handler, filter);
     return handler;
 }
@@ -2177,6 +2312,7 @@ MODULE_INIT(INIT_PRIORITY_JLOG)
     thePassLocalFilter = new PassLocalLogMsgFilter();
     thePassNoneFilter = new PassNoneLogMsgFilter();
     theStderrHandler = new HandleLogMsgHandlerTable(stderr, MSGFIELD_STANDARD);
+
     theSysLogEventLogger = new CSysLogEventLogger;
     theManager = new CLogMsgManager();
     theManager->resetMonitors();
@@ -2208,6 +2344,7 @@ static constexpr const char * logQueueDropAtt = "@queueDrop";
 static constexpr const char * logDisabledAtt = "@disabled";
 static constexpr const char * useSysLogpAtt ="@enableSysLog";
 static constexpr const char * capturePostMortemAtt ="@postMortem";
+static constexpr const char * logFormatAtt ="@format";
 
 static constexpr unsigned queueLenDefault = 512;
 static constexpr unsigned queueDropDefault = 0; // disabled by default
@@ -2223,6 +2360,32 @@ void setupContainerizedLogMsgHandler()
             removeLog();
             return;
         }
+
+        if (logConfig->hasProp(logFormatAtt))
+        {
+            const char *logFormat = logConfig->queryProp(logFormatAtt);
+            LOG(MCdebugInfo, "JLog: log format configuration detected '%s'!", logFormat);
+
+            bool newFormatDetected = false;
+            if (streq(logFormat, "xml"))
+            {
+                theStderrHandler = new HandleLogMsgHandlerXML(stderr, MSGFIELD_STANDARD);
+                newFormatDetected = true;
+            }
+            else if (streq(logFormat, "json"))
+            {
+                theStderrHandler = new HandleLogMsgHandlerJSON(stderr, MSGFIELD_STANDARD);
+                newFormatDetected = true;
+            }
+            else if (streq(logFormat, "table"))
+                LOG(MCdebugInfo, "JLog: default log format detected: '%s'!", logFormat);
+            else
+                LOG(MCoperatorWarning, "JLog: Invalid log format configuration detected '%s'!", logFormat);
+
+            if (newFormatDetected)
+                theManager->resetMonitors();
+        }
+
         if (logConfig->hasProp(logFieldsAtt))
         {
             //Supported logging fields: AUD,CLS,DET,MID,TIM,DAT,PID,TID,NOD,JOB,USE,SES,COD,MLT,MCT,NNT,COM,QUO,PFX,ALL,STD
@@ -2239,12 +2402,16 @@ void setupContainerizedLogMsgHandler()
             unsigned msgClasses = MSGCLS_all;
             const char *logClasses = logConfig->queryProp(logMsgClassesAtt);
             if (!isEmptyString(logClasses))
+            {
                 msgClasses = logMsgClassesFromAbbrevs(logClasses);
+            }
 
             unsigned msgAudiences = MSGAUD_all;
             const char *logAudiences = logConfig->queryProp(logMsgAudiencesAtt);
             if (!isEmptyString(logAudiences))
+            {
                 msgAudiences = logMsgAudsFromAbbrevs(logAudiences);
+            }
 
             const bool local = true; // Do not include remote messages from other components
             Owned<ILogMsgFilter> filter = getCategoryLogMsgFilter(msgAudiences, msgClasses, logDetail, local);
@@ -2902,7 +3069,7 @@ public:
                 lfs.set(fullFileSpec);
             else
                 lfs.set(logFileSpec.append(extension).str());
-            lmh = getFileLogMsgHandler(lfs.str(), NULL, msgFields, false);
+            lmh = getFileLogMsgHandler(lfs.str(), NULL, msgFields);
         }
         lmh->getLogName(expandedLogSpec);
         queryLogMsgManager()->addMonitorOwn( lmh, getCategoryLogMsgFilter(msgAudiences, msgClasses, maxDetail, local));

--- a/system/jlib/jlog.hpp
+++ b/system/jlib/jlog.hpp
@@ -585,6 +585,7 @@ public:
            unsigned port, LogMsgSessionId session)  __attribute__((format(printf,6, 0)));
     StringBuffer &            toStringPlain(StringBuffer & out, unsigned fields) const;
     StringBuffer &            toStringXML(StringBuffer & out, unsigned fields) const;
+    StringBuffer &            toStringJSON(StringBuffer & out, unsigned fields) const;
     StringBuffer &            toStringTable(StringBuffer & out, unsigned fields) const;
     static StringBuffer &     toStringTableHead(StringBuffer & out, unsigned fields);
     static void               fprintTableHead(FILE * handle, unsigned fields);
@@ -628,6 +629,13 @@ interface jlib_decl ILogMsgFilter : public IInterface
 };
 
 // Handler for log messages --- contains method to write or send messages
+
+typedef enum
+{
+    LOGFORMAT_xml,
+    LOGFORMAT_json,
+    LOGFORMAT_table
+} LogHandlerFormat;
 
 interface jlib_decl ILogMsgHandler : public IInterface
 {
@@ -772,8 +780,8 @@ extern jlib_decl ILogMsgFilter * getOrLogMsgFilter(ILogMsgFilter * arg1, ILogMsg
 extern jlib_decl ILogMsgFilter * getOrLogMsgFilterOwn(ILogMsgFilter * arg1, ILogMsgFilter * arg2);
 extern jlib_decl ILogMsgFilter * getSwitchLogMsgFilterOwn(ILogMsgFilter * switchFilter, ILogMsgFilter * yesFilter, ILogMsgFilter * noFilter);
 
-extern jlib_decl ILogMsgHandler * getHandleLogMsgHandler(FILE * handle = stderr, unsigned fields = MSGFIELD_all, bool writeXML = false);
-extern jlib_decl ILogMsgHandler * getFileLogMsgHandler(const char * filename, const char * headertext = 0, unsigned fields = MSGFIELD_all, bool writeXML = true, bool append = false, bool flushes = true);
+extern jlib_decl ILogMsgHandler * getHandleLogMsgHandler(FILE * handle = stderr, unsigned fields = MSGFIELD_all, LogHandlerFormat logFormat = LOGFORMAT_table);
+extern jlib_decl ILogMsgHandler * getFileLogMsgHandler(const char * filename, const char * headertext = 0, unsigned fields = MSGFIELD_all, LogHandlerFormat logFormat = LOGFORMAT_table, bool append = false, bool flushes = true);
 extern jlib_decl ILogMsgHandler * getRollingFileLogMsgHandler(const char * filebase, const char * fileextn, unsigned fields = MSGFIELD_all, bool append = false, bool flushes = true, const char *initialName = NULL, const char *alias = NULL, bool daily = false, long maxLogSize = 0);
 extern jlib_decl ILogMsgHandler * getBinLogMsgHandler(const char * filename, bool append = false);
 extern jlib_decl ILogMsgHandler * getPostMortemLogMsgHandler(const char * filebase, unsigned maxLinesToKeep, unsigned _messageFields=MSGFIELD_all);
@@ -784,9 +792,9 @@ extern jlib_decl void installLogMsgFilterSwitch(ILogMsgHandler * handler, ILogMs
 
 // Functions to make standard handlers and catagory filters and add to manager
 
-extern jlib_decl ILogMsgHandler * attachStandardFileLogMsgMonitor(const char * filename, const char * headertext = 0, unsigned fields = MSGFIELD_all, unsigned audiences = MSGAUD_all, unsigned classes = MSGCLS_all, LogMsgDetail detail = TopDetail, bool writeXML = true, bool append = false, bool flushes = true, bool local = false);
+extern jlib_decl ILogMsgHandler * attachStandardFileLogMsgMonitor(const char * filename, const char * headertext = 0, unsigned fields = MSGFIELD_all, unsigned audiences = MSGAUD_all, unsigned classes = MSGCLS_all, LogMsgDetail detail = TopDetail, LogHandlerFormat logFormat = LOGFORMAT_table, bool append = false, bool flushes = true, bool local = false);
 extern jlib_decl ILogMsgHandler * attachStandardBinLogMsgMonitor(const char * filename, unsigned audiences = MSGAUD_all, unsigned classes = MSGCLS_all, LogMsgDetail detail = TopDetail, bool append = false, bool local = false);
-extern jlib_decl ILogMsgHandler * attachStandardHandleLogMsgMonitor(FILE * handle = stderr, unsigned fields = MSGFIELD_all, unsigned audiences = MSGAUD_all, unsigned classes = MSGCLS_all, LogMsgDetail detail = TopDetail, bool writeXML = false, bool local = false);
+extern jlib_decl ILogMsgHandler * attachStandardHandleLogMsgMonitor(FILE * handle = stderr, unsigned fields = MSGFIELD_all, unsigned audiences = MSGAUD_all, unsigned classes = MSGCLS_all, LogMsgDetail detail = TopDetail, LogHandlerFormat logFormat = LOGFORMAT_table, bool local = false);
 
 // Function to construct filter from serialized and XML forms, and construct handler from XML form, and attach monitor(s) from XML form
 

--- a/system/jlib/jlog.ipp
+++ b/system/jlib/jlog.ipp
@@ -445,6 +445,17 @@ protected:
     StringBuffer              curMsgText;
 };
 
+class HandleLogMsgHandlerJSON : implements HandleLogMsgHandler, public CInterface
+{
+public:
+    HandleLogMsgHandlerJSON(FILE * _handle, unsigned _fields) : HandleLogMsgHandler(_handle, _fields) {}
+    IMPLEMENT_IINTERFACE;
+    void                      handleMessage(const LogMsg & msg);
+    bool                      needsPrep() const { return false; }
+    void                      prep() {}
+    void                      addToPTree(IPropertyTree * tree) const;
+};
+
 class HandleLogMsgHandlerXML : implements HandleLogMsgHandler, public CInterface
 {
 public:
@@ -491,6 +502,17 @@ protected:
     bool                      flushes;
     mutable CriticalSection   crit;
     StringBuffer              curMsgText;
+};
+
+class FileLogMsgHandlerJSON : implements FileLogMsgHandler, public CInterface
+{
+public:
+    FileLogMsgHandlerJSON(const char * _filename, const char * _headerText = 0, unsigned _fields = MSGFIELD_all, bool _append = false, bool _flushes = true) : FileLogMsgHandler(_filename, _headerText, _fields, _append, _flushes) {}
+    IMPLEMENT_IINTERFACE;
+    void                      handleMessage(const LogMsg & msg);
+    bool                      needsPrep() const { return false; }
+    void                      prep() {}
+    void                      addToPTree(IPropertyTree * tree) const;
 };
 
 class FileLogMsgHandlerXML : implements FileLogMsgHandler, public CInterface

